src/tools/crawler/main.rs

```diff
@@ -39,8 +40,12 @@ const LOG_PATH: &str = "crawler-log.txt";
 #[clap(author, version, about, long_about = None)]
 struct Args {
```

Comment:
How about using an `ArgGroup` to make sure that at least one of the source addrs arguments is given?
https://docs.rs/clap/latest/clap/struct.ArgGroup.html#method.required

This should do the job:
```suggestion
#[clap(author, version, about, long_about = None)]
#[clap(group(
        ArgGroup::new("source_addrs")
            .required(true)
            .args(&["seed-addrs", "dns_seed"])
        ))],
struct Args {
```

Reply:
Nice one!
Exact form would be like that:

```
#[clap(group(
ArgGroup::new("source_addrs")
.required(true)
.multiple(true)
.args(&["seed_addrs", "dns_seed"])
))]
```

I've added `.multiple(true)` which allows to set both dns seeder and give just peer 😄